R/workplan.R

```
@@ -149,3 +156,133 @@ drake_plan_override <- function(target, field, config){
+}
+
+#' @title Declare the file outputs of a workflow plan command.
+#' @description Use this function to help write the commands
+#'   in your workflow plan data frame. You can only specify
+#'   one file output per command. See the examples
+#'   for a full explanation.
+#' @export
+#' @seealso file_input knitr_input
+#' @return A character string, the file path of the file output.
+#' @param path Character string of length 1. File path
+#'   of the file output of a command in your
+#'   workflow plan data frame.
+#' @param ... Do not use. For informative input handling only.
+#' @examples
+#' \dontrun{
+#' test_with_dir("Contain side effects", {
+#' # The `file_output()` and `file_input()` functions
+#' # just takes in strings and returns them.
+#' file_output("summaries.txt")
+#' # Their main purpose is to orchestrate your custom files
+#' # in your workflow plan data frame.
+#' suppressWarnings(
+#'   plan <- drake_plan(
+#'     write.csv(mtcars, file_output("mtcars.csv")),
+#'     contents = read.csv(file_input("mtcars.csv")),
+#'     strings_in_dots = "literals" # deprecated but useful: no single quotes needed. # nolint
+#'   )
+#' )
+#' plan
+#' # Drake knows "\"mtcars.csv\"" is the first target
+#' # and a dependency of `contents`. See for yourself:
+#' config <- make(plan)
+#' file.exists("mtcars.csv")
+#' vis_drake_graph(config)
+#' # See also `knitr_input()`. `knitr_input()` is like `file_input()`
+#' # except that it analyzes active code chunks in your `knitr`
+#' # source file and detects non-file dependencies.
+#' # That way, updates to the right dependencies trigger rebuilds
+#' # in your report.
+#' })
+#' }
+file_output <- function(path, ...){
+  path <- c(path, as.character(c(...)))
+  if (length(path) != 1){
+    warning(
+      "In file_output(), the `path` argument must ",
+      "have length 1. Supplied length = ", length(path), ". ",
+      "using first file output: ", path[1], ".",
+      call. = FALSE
+    )
+  }
+  as.character(path[1])
```

This still gives wrong results if `path` is a list, e.g. `file_output(list(a = 1, b = 2))`. Can we just get rid of the `...` argument and fail early if the user tries to supply a second argument?

---

I assumed `file_output(list(a = 1, b = 2))` return `"1"` (current behavior). What do you think it should return?

Your comment actually spurred me to refactor the code analysis. I will have another PR for your review soon.